### PR TITLE
Update stock brushes, fonts, stock icons, and add resource loader

### PR DESCRIPTION
**Pull request by dblohm7**
* We update win to pick up the newest constants;
* We add white and black stock brushes;
* We change the stock brush code to use GetStockObject;
* We add Light and SemiLight font weights;
* We change stock icon functions to consist of default and small sizes;
* We add functions for directly loading resources.

### CODE REVIEW

File: brush.go [hunk 1]
```diff
@@ -120,54 +130,55 @@ func (bb *brushBase) detachWindow(wb *WindowBase) {
...
-	}
-
-	b.brushBase.Dispose()
+func (sb *stockBrush) Dispose() {
+	// No-op for stock brushes
 }
 
-func (*nullBrush) logbrush() *win.LOGBRUSH {
-	return &win.LOGBRUSH{LbStyle: win.BS_NULL}
+func (sb *stockBrush) simple() bool {
```
Comment (nickkhyl) [inline brush.go:141]: nit: use a value receiver?

File: font.go [hunk 2]
```diff
@@ -57,11 +63,11 @@ type Font struct {
...
 // NewFont returns a new Font with the specified attributes.
 func NewFont(family string, pointSize int, style FontStyle) (*Font, error) {
-	if style > FontSemiBold|FontBold|FontItalic|FontUnderline|FontStrikeOut {
+	if style > FontSemiLight|FontLight|FontSemiBold|FontBold|FontItalic|FontUnderline|FontStrikeOut {
 		return nil, newError("invalid style")
 	}
-	if (style & (FontSemiBold | FontBold)) == FontSemiBold|FontBold {
-		return nil, newError("invalid style: semi-bold and bold are mutually-exclusive")
+	if bits.OnesCount8(byte(style&(FontLight|FontSemiLight|FontSemiBold|FontBold))) > 1 {
+		return nil, newError("invalid style: font weights are mutually-exclusive")
```
Comment (nickkhyl) [inline font.go:70]:
```suggestion
	if bits.OnesCount8(byte(style&(fontWeightBits))) > 1 {
		return nil, newError("invalid style: font weights are mutually-exclusive")
```
?

File: font.go [hunk 3]
```diff
@@ -57,11 +63,11 @@ type Font struct {
 
 // NewFont returns a new Font with the specified attributes.
 func NewFont(family string, pointSize int, style FontStyle) (*Font, error) {
-	if style > FontSemiBold|FontBold|FontItalic|FontUnderline|FontStrikeOut {
+	if style > FontSemiLight|FontLight|FontSemiBold|FontBold|FontItalic|FontUnderline|FontStrikeOut {
```
Comment (nickkhyl) [inline font.go:66]: maybe make this a const similar to `fontWeightBits`?

File: font.go [hunk 4]
```diff
@@ -85,6 +91,8 @@ func NewFont(family string, pointSize int, style FontStyle) (*Font, error) {
 	return font, nil
 }
 
+const fontWeightSemiLight = 350
```
Comment (nickkhyl) [inline font.go:94]: Maybe this?
```suggestion
const fontWeightSemiLight = (win.FW_LIGHT + win.FW_NORMAL) / 2
```
It's not super clear where the magic number of 350 comes from.

File: resource.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+// Bytes returns the resource's data as a byte slice.
+func (res *Resource) Bytes() []byte {
+	return res.data
+}
+
+// Stream returns the resource's data as a COM Stream.
+func (res *Resource) Stream() (com.Stream, error) {
+	return com.NewMemoryStream(res.data)
+}
```
Comment (nickkhyl) [inline resource.go:32]: use value receivers here?

File: resource.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+// LoadResourceByID locates the resource identified by id and resType
+// from the current process's executable binary and returns its contents
```
Comment (nickkhyl) [inline resource.go:35]: Just wanted to double-check that we don't have plans to use this from DLLs, such as the MSI custom actions.

File: resource.go [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+// LoadCustomResourceByName locates the resource identified by name
+// from the current process's executable binary and returns its contents
+// as a *Resource.
+func LoadCustomResourceByName(name string) (*Resource, error) {
+	return LoadResourceByName(name, win.RT_RCDATA)
+}
+
+func loadResource(name *uint16, resType win.ResourceType) (result *Resource, err error) {
+	hres := win.FindResource(0, name, win.MAKEINTRESOURCE(uint16(resType)))
```
Comment (nickkhyl) [inline resource.go:68]: We probably don't need the `uint16` conversion here.

File: resource.go [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/dblohm7/wingoes/com"
+	"github.com/tailscale/win"
+	"golang.org/x/exp/constraints"
+	"golang.org/x/sys/windows"
+)
+
+// Resource encapsulates data loaded from the executable's Win32 resources.
+type Resource struct {
+	data []byte
+}
```
Comment (nickkhyl) [inline resource.go:22]:
```suggestion
// Resource encapsulates data loaded from the executable's Win32 resources.
type Resource []byte
```
?

If you'd like to encapsulate it, that's okay too. But then maybe update the functions below to return `(Resource, error)`?
There's probably no good reason to return a pointer to a slice, and this is essentially a value type.

**Review by nickkhyl — APPROVED** [review]
LGTM. Just a few optional things to consider.

File: resource.go (hunk 6 above)
Comment (dblohm7) [reply resource.go:35]: Nope, I'd say in general we're not gonna use `walk` stuff from there.

File: resource.go (hunk 8 above)
Comment (nickkhyl) [reply resource.go:22]: Actually, I think the functions should return `(Resource, error)` regardless of whether we make this a named slice type.

**Pull request merged**